artifact/v3/manifest_sig/manifest_sig.cpp

```diff
@@ -39,8 +41,21 @@ ExpectedManifestSignature Parse(io::Reader &reader) {
 }
 
 expected::ExpectedBool VerifySignature(
-	ManifestSignature &signature, vector<string> &artifact_verify_keys) {
-	return expected::ExpectedBool(true);
+	ManifestSignature &signature, vector<uint8_t> raw_data, vector<string> &artifact_verify_keys) {
+	error::Error err;
+	for (auto key : artifact_verify_keys) {
+		auto e_verify_sign = crypto::VerifySign(key, raw_data, signature.value);
+		if (e_verify_sign && e_verify_sign.value()) {
+			return expected::ExpectedBool(true);
+		}
+		if (!e_verify_sign) {
+			err = err.FollowedBy(e_verify_sign.error());
+		}
+	}
+	if (err == error::NoError) {
+		return expected::ExpectedBool(false);
```

Comment:
Hmm, this will fail signature verification if one key isn't found for example, even if one of the other keys work. I could see this happening with a broken symlink for example. Is it better to let a valid verification trump everything else? I don't see how it would be possible to "accidentally" verify something. If it's verified, it's verified, no matter the state of the rest of the keys.

Reply:
Mmmm. I didn't intend that behavior, but maybe I expressed it incorrectly.

So inside the for loop we'll return `true` when the first key is found (parsed, etc) and validates correctly. We explicitly check it with `if (e_verify_sign && e_verify_sign.value())`. Otherwise we just append errors (if exist).

Finally, this part that you are pointing out (which is outside of the loop) will either return just a false if there were no errors but still no correct verification, or the concatenated errors (line below).

Does it make sense?

Reply:
You're right, I misread it. Looks good! 👍